### PR TITLE
DRILL-8354: Add IS_EMPTY Function

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-8354](https://issues.apache.org/jira/browse/DRILL-8354): Add IS_EMPTY Function

## Description
When analyzing data, there is currently no single function to evaluate whether a given field is empty.  With scalar fields, this can be accomplished with the `IS NOT NULL` operator, but with complex fields, this is more challenging as complex fields are never null. 
This PR adds a UDF called `IS_EMPTY()` which accepts any type of field and returns true if the field does not contain data.  

In the case of scalar fields, if the field is `null` this returns true.  In the case of complex fields, which can never be `null`, in the case of lists, the function returns true if the list is empty.  In the case of maps, it returns true if all of the map's fields are unpopulated. 

## Documentation
I will update the documentation once this is merged.

## Testing
Added unit tests.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/IsEmptyUtils.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+   * This function recursively traverses a Drill map to determine whether the map is empty or not.
+   * @param reader A {@link FieldReader} containing the field in question
+   * @return True if the field contains no data, false if it does.
+   */
+  public static boolean mapIsEmpty(FieldReader reader) {
+
+    if (reader.getType().getMinorType() == MinorType.MAP) {
+      SingleMapReaderImpl mapReader = (SingleMapReaderImpl) reader;
+
+      // If the map reader has no fields returns nothing return true
```
Comment (jnturton) [inline exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/IsEmptyUtils.java:37]:
```suggestion
      // If the map reader has no fields return true
```

File: exec/java-exec/src/test/resources/jsoninput/is_empty_tests.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "numeric_col": 2.3,
+    "text_col": "",
+    "list_col": [],
+    "map_column": {}
+  },{
+    "numeric_col": null,
+    "text_col": null,
+    "list_col": [],
+    "map_column": {}
+  }
```
Comment (jnturton) [inline exec/java-exec/src/test/resources/jsoninput/is_empty_tests.json:26]:
```suggestion
  }, {
    "numeric_col": 1
  }
```
Can we add an object that is outright missing some queried properties and check that the is_empty unit tests pick those up as being empty?

**Review by jnturton — CHANGES_REQUESTED** [review]

File: exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/IsEmptyUtils.java (hunk 1 above)
Comment (cgivre) [reply exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/IsEmptyUtils.java:37]: Fixed.

File: exec/java-exec/src/test/resources/jsoninput/is_empty_tests.json (hunk 2 above)
Comment (cgivre) [reply exec/java-exec/src/test/resources/jsoninput/is_empty_tests.json:28]: @jnturton Done!  I addressed your review comments.  Thanks for the quick review!
